### ISSUE TITLE
Sprint plan for OSIO Build: # 147

### ISSUE THREAD

**Issue by krishnapaparaju** [openshiftio/openshift.io#2779]
**Work item tracker**
   https://github.com/orgs/fabric8io/projects/2

**Integration testing**

- [ ] OSIO tenant space

- [ ] Jenkins Proxy

- [ ] Jenkins Idler

**Collect metrics around OSIO builds** #2778
- [ ]  An instance is to be idled, is actually getting idled ?

- [ ] An instance is to be unidled, is actually getting unidled ?

- [ ] How many times web hooks are being received, store the source for these webhooks (& check if build / required action got started with success)

**Tenant Jenkins**

- [x] https://github.com/openshiftio/openshift.io/issues/2608 - Tenant Jenkins boots up with way too many config issues. 
- [ ] https://github.com/openshiftio/openshift.io/issues/2729 - Do not start parallel builds for a OSIO user, notify user

- [ ] #2384 - Jenkins trying to spin up multiple slaves at the same time  

- [ ] TODO add more 

**Pipeline library**

- [ ] https://github.com/openshiftio/openshift.io/issues/2825  - Fix the readonly mount + chmod issue across pipeline library

**UI**

- [ ] Incorrect  build URLs
        https://github.com/openshiftio/openshift.io/issues/2744
          
**Proxy**

- [ ] Replay incoming requests, might be part of Metrics collection work 
       https://github.com/fabric8-services/fabric8-jenkins-proxy/issues/201

- [ ] Code refactoring and Add Unit Tests
       https://github.com/fabric8-services/fabric8-jenkins-proxy/issues/179

- [ ] Have incoming webhooks at a queue, for better visibility, scale with more proxies , plays into collecting metrics work
      https://github.com/fabric8-services/fabric8-jenkins-proxy/issues/175

- [ ]  https://github.com/fabric8-services/fabric8-jenkins-proxy/issues/179

- [ ] Integration tests https://github.com/fabric8-services/fabric8-jenkins-proxy/issues/208

- [ ] Jenkins Proxy load testing  https://github.com/fabric8-services/fabric8-jenkins-proxy/issues/214

**Idler**

- [ ]  https://github.com/fabric8-services/fabric8-jenkins-idler/milestone/1

- [ ] Estimate memory requirements for Idler
       https://github.com/fabric8-services/fabric8-jenkins-idler/issues/161

- [ ] Idler retaining state
      https://github.com/fabric8-services/fabric8-jenkins-idler/issues/143

- [ ] Build would not start for OISO user
       https://github.com/openshiftio/openshift.io/issues/2600

- [ ] Build pipeline fails
         https://github.com/openshiftio/openshift.io/issues/2690

- [ ] Integration tests 
         https://github.com/fabric8-services/fabric8-jenkins-idler/issues/176

**Auth**

- [ ] Intermittent auth issue to reach jenkins.openshift.io 
       https://github.com/fabric8-services/fabric8-jenkins-proxy/issues/205

**Fabric8 Builds**

- [ ] 

- [ ] 

**Fabric8 Maven Plugin**

- [ ] https://github.com/fabric8io/fabric8-maven-plugin/issues/1245
- [ ] https://github.com/fabric8io/fabric8-maven-plugin/issues/1093
- [ ] https://github.com/fabric8io/fabric8-maven-plugin/issues/1243
- [ ] https://github.com/fabric8io/fabric8-maven-plugin/issues/1198

**Fabric8 Kubernetes Java Client**

- [ ] https://github.com/fabric8io/kubernetes-client/issues/989
- [ ] https://github.com/fabric8io/kubernetes-client/issues/1035
- [x] https://github.com/fabric8io/kubernetes-client/issues/957
- [ ] https://github.com/fabric8io/kubernetes-client/issues/1046

> I also see that you have a lot of issues outside of openshift.io repo. You can move them to this repo with https://github-issue-mover.appspot.com/. We need them all in one place so that we can track them.

**Comment by pradeepto** [comment]
@krishnapaparaju 

FMP and FKC are also owned by build team. Those are upstream projects that we maintain for now and make sure they work for OpenShift and OSIO bits. We need to add the following issues to the sprint that @pgarg and @rohanKanojia will take care of.

## Fabric8 Maven Plugin

https://github.com/fabric8io/fabric8-maven-plugin/issues/1245
https://github.com/fabric8io/fabric8-maven-plugin/issues/1093
https://github.com/fabric8io/fabric8-maven-plugin/issues/1243

## Fabric8 Kubernetes Plugin

https://github.com/fabric8io/kubernetes-client/issues/1035
https://github.com/fabric8io/kubernetes-client/issues/957
https://github.com/fabric8io/kubernetes-client/issues/1005

**Comment by pradeepto** [comment]
@krishnapaparaju Echoing what I said on MM. There are far too many proxy and idler issues in the list above. There are two people who work on that (by design) - @chmouel and @kishansagathiya . We can't possibly doing 10+ issues of those in one sprint. We need to reduce those.

**Comment by chmouel** [comment]
We need tasks for Jenkins and Builds, I would advise strongly to look over the backlogs on https://github.com/orgs/fabric8io/projects/2

**Comment by pradeepto** [comment]
Duplicate of https://github.com/openshiftio/openshift.io/issues/2245 ?

**Comment by joshuawilson** [comment]
@krishnapaparaju To avoid more then one person working on an issue, please assign the person working on it and set the Milestone Sprint ###. At least one of these was picked up in the Platform planning. 

I also see that you have a lot of issues outside of openshift.io repo. You can move them to this repo with https://github-issue-mover.appspot.com/. We need them all in one place so that we can track them.

**Comment by maxandersen** [comment]
> I also see that you have a lot of issues outside of openshift.io repo. You can move them to this repo with https://github-issue-mover.appspot.com/. We need them all in one place so that we can track them.

what issues is it you believe should be moved over ?

**Comment by joshuawilson** [comment]
@maxandersen if you look at the main description, there are a lot of links to external repo. i.e. fabric8-services/fabric8-jenkins-proxy#201

**Comment by chmouel** [comment]
@joshuawilson what's the point of having to move all the issues from the internal repositories to `openshiftio` ?

**Comment by joshuawilson** [comment]
@chmouel everything is being tracked by PM in https://github.com/openshiftio/openshift.io/projects/3

And if everything is in one place we can export it to planner (when we are ready to).

**Comment by kishansagathiya** [comment]
This issue was created for sprint planning. I don't see a reason to keep it open once the sprint is done. 
Closing it.

**Comment by bartoszmajsak** [comment]
@kishansagathiya before closing it would be good to update the status of the checklist in the description
